main.rb

```diff
@@ -103,127 +224,123 @@ def update_project(project, file_response, access_token)
+    req.body = JSON.dump(dict)
+    res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
+      http.request(req)
+    end
+
+    if (res.kind_of? Net::HTTPSuccess)
+      puts('Project updated successfully...')
+      return get_parsed_response(res.body)
+    elsif (res.kind_of? Net::HTTPClientError)
+      puts(get_parsed_response(res.body)[:message])
+      count += 1
+    else
+      puts("Error while updating Project. Response from server: #{get_parsed_response(res.body)}")
+      count += 1
+    end
   end
-  JSON.parse(res.body, symbolize_names: true)
+  exit(1)
 end
 
 def start(access_token)
-  uri = URI.parse("https://testinium.io/Testinium.RestApi/api/plans/#{$plan_id}/run")
-  req = Net::HTTP::Get.new(uri.request_uri,
-                           { 'Content-Type' => 'application/json', 'Authorization' => "Bearer #{access_token}", 'current-company-id' => "#{$company_id}"})
-  res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
-    http.request(req)
-  end
-  case res
-  when Net::HTTPClientError, Net::HTTPServerError
-    puts(res.body)
-    puts "\nError starting plan: #{res.code} (#{res.message})\n\n"
-    exit(1)
+  is_success = false
+  count = 1
+
+  while !is_success && is_count_less_than_max_api_retry(count) do
+    check_timeout()
+    puts("Starting a new test plan... Number of attempts: #{count}")
+    uri = URI.parse("https://testinium.io/Testinium.RestApi/api/plans/#{$plan_id}/run")
+    req = Net::HTTP::Get.new(uri.request_uri,
+                             { 'Content-Type' => 'application/json', 'Authorization' => "Bearer #{access_token}", 'current-company-id' => "#{$company_id}" })
+    res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
+      http.request(req)
+    end
+
+    if (res.kind_of? Net::HTTPSuccess)
+      puts('Plan started successfully...')
+      return get_parsed_response(res.body)[:execution_id]
+    elsif (res.kind_of? Net::HTTPClientError)
+      puts(get_parsed_response(res.body)[:message])
+      count += 1
+    else
+      puts("Error while starting Plan. Response from server: #{get_parsed_response(res.body)}")
+      count += 1
+    end
   end
-  JSON.parse(res.body, symbolize_names: true)[:execution_id]
+  exit(1)
 end
 
 def get_report(execution_id, access_token)
-  uri = URI.parse("https://testinium.io/Testinium.RestApi/api/executions/#{execution_id}")
-  req = Net::HTTP::Get.new(uri.request_uri,
-                           { 'Content-Type' => 'application/json', 'Authorization' => "Bearer #{access_token}" })
-  res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
-    http.request(req)
-  end
-  case res
-  when Net::HTTPClientError, Net::HTTPServerError
-    puts(res.body)
-    puts "\nError getting report: #{res.code} (#{res.message})\n\n"
-    exit(1)
-  end
+  is_success = false
```

Comment:
We do not need the `is_success` variable. "get_report()" function will work fine without it.